sopel/modules/find.py

```diff
@@ -163,10 +163,14 @@ def findandreplace(bot, trigger):
 
     sep = trigger.group('sep')
     old = trigger.group('old').replace('\\%s' % sep, sep)
-    new = bold(trigger.group('new')).replace('\\%s' % sep, sep)
+    new = trigger.group('new')
     me = False  # /me command
     flags = trigger.group('flags') or ''
 
+    # only clean/format the new string if it's non-empty
+    if new:
```

Comment:
I assume you don't want to have `    ` between bold either? Then I'd strip the string just for the condition:

```suggestion
    if new.strip():
```

Reply:
I should test this with formatting codes in the replacement. IIRC `str.strip()` will remove some (but not all) of those.

Reply:
As I thought:

```pycon
>>> from sopel import formatting
>>> for f in formatting.CONTROL_FORMATTING:
...     print(f.encode('unicode_escape'), f.strip().encode('unicode_escape'))
...
b'\\x0f' b'\\x0f'
b'\\x03' b'\\x03'
b'\\x04' b'\\x04'
b'\\x02' b'\\x02'
b'\\x1d' b''
b'\\x1f' b''
b'\\x1e' b''
b'\\x11' b'\\x11'
b'\\x16' b'\\x16'
```

So here again we run into the same issue with `str.strip()` that led to [the `_format_safe()` method in `choose`](https://github.com/sopel-irc/sopel/blob/b04c285d4fd780f8a70d738ad556093ac327770c/sopel/modules/choose.py#L26): its default set of characters to strip will remove _some_ (but not all) IRC formatting bytes. Of course it's not being used for output, but it would still cause some inconsistent behavior in the `find` plugin's output with certain user inputs.

Perhaps a better approach would be to skip applying bold formatting if the replacement contains _any_ formatting bytes? Otherwise yes, stripping it of whitespace would be OK here.

(Of course the idea of a monkey-patch to `str` here that adds some `str.safe_strip()` method crossed my mind… but there are _many_ reasons that's a bad idea, chief among them being that Sopel shouldn't mess with stdlib types.)

Reply:
I guess edge cases of an edge case are too edgy for today.